src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportBasicService.java

```diff
@@ -0,0 +1,351 @@
+    // IMPORTANT: the transactional context only works if you invoke this method from another class
+    public ProgrammingExercise importProgrammingExerciseBasis(final ProgrammingExercise templateExercise, final ProgrammingExercise newExercise) {
+        // Set values we don't want to copy to null
+        setupExerciseForImport(newExercise);
+        newExercise.setBranch(versionControlService.get().getDefaultBranchOfArtemis());
+
+        // Note: same order as when creating an exercise
+        programmingExerciseParticipationService.setupInitialTemplateParticipation(newExercise);
+        programmingExerciseParticipationService.setupInitialSolutionParticipation(newExercise);
+        setupTestRepository(newExercise);
+        programmingExerciseService.initParticipations(newExercise);
+
+        // Hints, tasks, test cases and static code analysis categories
+        Map<Long, Long> newHintIdByOldId = exerciseHintService.copyExerciseHints(templateExercise, newExercise);
+        programmingExerciseRepository.save(newExercise);
+        Map<Long, Long> newTestCaseIdByOldId = importTestCases(templateExercise, newExercise);
+        Map<Long, Long> newTaskIdByOldId = importTasks(templateExercise, newExercise, newTestCaseIdByOldId);
+        updateTaskExerciseHintReferences(templateExercise, newExercise, newTaskIdByOldId, newHintIdByOldId);
+        importSolutionEntries(templateExercise, newExercise, newTestCaseIdByOldId, newHintIdByOldId);
+
+        // Copy or create SCA categories
+        if (Boolean.TRUE.equals(newExercise.isStaticCodeAnalysisEnabled() && Boolean.TRUE.equals(templateExercise.isStaticCodeAnalysisEnabled()))) {
+            importStaticCodeAnalysisCategories(templateExercise, newExercise);
+        }
+        else if (Boolean.TRUE.equals(newExercise.isStaticCodeAnalysisEnabled()) && !Boolean.TRUE.equals(templateExercise.isStaticCodeAnalysisEnabled())) {
+            staticCodeAnalysisService.createDefaultCategories(newExercise);
+        }
+
+        // An exam exercise can only be in individual mode
+        if (newExercise.isExamExercise()) {
+            newExercise.setMode(ExerciseMode.INDIVIDUAL);
+            newExercise.setTeamAssignmentConfig(null);
+        }
+
+        importSubmissionPolicy(newExercise);
+
+        // Re-adding auxiliary repositories
+        List<AuxiliaryRepository> auxiliaryRepositoriesToBeImported = templateExercise.getAuxiliaryRepositories();
+
+        for (AuxiliaryRepository auxiliaryRepository : auxiliaryRepositoriesToBeImported) {
+            AuxiliaryRepository newAuxiliaryRepository = auxiliaryRepository.cloneObjectForNewExercise();
+            auxiliaryRepositoryRepository.save(newAuxiliaryRepository);
+            newExercise.addAuxiliaryRepository(newAuxiliaryRepository);
+        }
+
+        programmingExerciseRepository.save(newExercise);
+
+        return newExercise;
```

Comment:
```suggestion
        return programmingExerciseRepository.save(newExercise);
```

Would this work or does some LAZY attributes are missing then? If yes, a comment would help.